tools/wptrunner/wptrunner/tests/test_testloader.py

```diff
@@ -135,6 +168,110 @@ def test_filter_unicode():
         TestFilter(manifest_path=f.name, test_manifests=tests)
 
 
+def test_filter_directory(manifest1):
+    with tempfile.TemporaryDirectory() as tmp:
+        include = os.path.normpath(os.path.join(tmp, "a", "b"))
+        os.makedirs(include)
+        test_filter = TestFilter({manifest1: {"tests_path": tmp}}, include=[include])
+        items = sorted(test_filter(manifest1), key=lambda item: item[1])
+    assert len(items) == 2
+    test_type, test_path, tests = items[0]
+    assert test_type == "testharness"
+    assert test_path == os.path.join("a", "b", "bar.html")
+    assert {test.id for test in tests} == {"/a/b/bar.html"}
+    test_type, test_path, tests = items[1]
+    assert test_type == "testharness"
+    assert test_path == os.path.join("a", "b", "c.html")
+    assert {test.id for test in tests} == {"/a/b/c.html"}
+
+
+def test_filter_variants_for_scope(manifest1):
```

Comment:
This test covers https://github.com/web-platform-tests/wpt/pull/42377#discussion_r1377717383.